### PR TITLE
Editorial changes to columns names and ids (part 2)

### PR DESCRIPTION
**Pull request by ijurica**
_The changes in this PR are in accordance with the agreement that we will continue to use the previous version of the editorial guidelines in FOCUS version 1.1 (for details, see the table below)._

<table>
    <tr>
        <th width="20%">Component</th>
        <th width="30%">Markdown (examples)</th>
        <th width="50%">Editorial Guidelines</th>
    </tr>
    <tr>
        <td><strong>Column Names</strong>:</td>
        <td>
            <strong>Column Names</strong>:<br>
            Pricing Quantity<br>
            Pricing Unit<br>
            Provider<br>
        </td>
        <td>
            <ul>
                <li>Normal text without bold or italics.</li>
                <li>Use the display name in the introductory (non-normative) section and description.</li>
                <li>The first occurrence in a section is linked to the section, except when referring to itself.</li>
            </ul>
        </td>
    </tr>
    <tr>
        <td><strong>Attribute Names</strong>:</td>
        <td>
            <strong>Attribute Names</strong>:<br>
             Currency Code Format<br>
            Date/Time Format<br>
        </td>
        <td>
            <ul>
                <li>Normal text without bold or italics.</li>
                <li>Use the attribute display name in the introductory (non-normative) section, description, normative section, and content constraints table.</li>
                <li>The first occurrence in a section is linked to the section, except when referring to itself.</li>
            </ul>
        </td>
    </tr>
    <tr>
        <td><strong>Column IDs</strong>:</td>        
        <td>
            <strong>Column IDs</strong>:<br>
            PricingQuantity<br>
            PricingUnit<br>
            ProviderName<br>
        </td>
        <td>
            <ul>
                <li>Normal text without bold or italics.</li>
                <li>Use the column ID in the normative section and when referencing column values.</li>
                <li>For column IDs, use PascalCase (the first letter of every word is capitalized).</li>
                <li>The first occurrence in a section is linked to the corresponding column, except when referring to itself.</li>
            </ul>
        </td>
    </tr>
    <tr>
        <td><strong>Glossary</strong></td>
        <td>
            [*SKU*](#glossary:sku) <br>
            [*resource*](#glossary:resource) <br>
            [*service*](#glossary:service) <br>
        </td>
        <td>
            <ul>
                <li>Italic.</li>
                <li>The first occurrence in a section is linked to the glossary term.</li>
            </ul>
        </td>
    </tr>
</table>

---

**Types of Editorial Changes:**

- Aligned the **formatting** of references to columns (display name, ID) and glossary terms with the approach used in other columns.
- Aligned **formatting of references to format attributes** in normative requirements, opting for the **attribute name instead of the attribute ID** (_see Note for reasoning_):
  - CommitmentDiscountUnit, ConsumedUnit and PricingUnit addressed in this PR
  - SkuPriceDetails addressed in PR #588 
- Addressed several **missing and broken links** to columns and glossary terms.

**Changes applied to the following columns:**

- Commitment Discount Quantity
- Commitment Discount Unit
- Consumed Unit
- Pricing Unit
- Service Subcategory

---

**Notes:** 

- I prefer using the attribute name over the attribute ID when referencing formatting attributes, at least for now, considering:
  - The attribute name was used in the Content Constraints tables in all instances.
  - The attribute name was used in the Normative Requirements in 13 instances.
  - The attribute ID was used in the Normative Requirements in only 4 instances (CommitmentDiscountUnit, ConsumedUnit, PricingUnit, SkuPriceDetails).
  
- In 2 instances (ChargePeriodStart and ChargePeriodEnd), although listed in the constraints table, the requirement to follow the Date/Time format wasn’t explicitly mentioned in the normative requirements paragraph. I suggest addressing all Date/Time references in version 1.2 or later.

### CODE REVIEW

File: specification/columns/pricingunit.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 
-The PricingUnit column MUST be present in a FOCUS dataset. This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
+The PricingUnit column MUST be present in a FOCUS dataset. This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [Unit Format](#unitformat) attribute.
```
Comment (ijurica) [inline specification/columns/pricingunit.md:5]: Adding a link to the FOCUS dataset term in the glossary (see PR #592 for linking to the glossary term)

```suggestion
The PricingUnit column MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset). This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [Unit Format](#unitformat) attribute.
```

**Review by cnharris10 — APPROVED** [review]

File: specification/columns/consumedunit.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ The Consumed Unit represents a provider-specified measurement unit indicating ho
 The ConsumedUnit column adheres to the following requirements:
 
 * ConsumedUnit MUST be present in the billing data when the provider supports the measurement of usage.
-* ConsumedUnit MUST be of type String, and the units of measure used in ConsumedUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
+* ConsumedUnit MUST be of type String, and the units of measure used in ConsumedUnit SHOULD adhere to the values and format requirements specified in the [Unit Format](#unitformat) attribute.
```
Comment (udam-f2) [inline specification/columns/consumedunit.md:8]: should be normative text therefore attribute id not attribute name?

Comment (ijurica) [reply specification/columns/consumedunit.md:8]: **Heads up:** 
For the sake of consistency, and based on the discussion in the maintainers' meeting on Oct 8th, we'll be switching **from attribute names to attribute IDs in the normative paragraphs**, while retaining attribute names in the Content Constraints tables.

File: specification/columns/commitmentdiscountquantity.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+When [CommitmentDiscountCategory](#commitmentdiscountcategory) is "Usage" (usage-based *commitment discounts*), the Commitment Discount Quantity reflects the predefined amount of usage purchased or consumed. If [*commitment discount flexibility*](#glossary:commitment-discount-flexibility) is applicable, this value may be further transformed based on additional, provider-specific requirements. When CommitmentDiscountCategory is "Spend" (spend-based *commitment discounts*), the Commitment Discount Quantity reflects the predefined amount of spend purchased or consumed.
 
 The CommitmentDiscountQuantity column adheres to the following requirements:
 
 * CommitmentDiscountQuantity MUST be present in a FOCUS dataset when the provider supports *commitment discounts*.
 * CommitmentDiscountQuantity MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements.
```
Comment (ijurica) [inline specification/columns/commitmentdiscountquantity.md:10]:
```suggestion
* CommitmentDiscountQuantity MUST be of type Decimal and MUST conform to [NumericFormat](#numericformat) requirements.
```

File: specification/columns/commitmentdiscountunit.md [hunk 4]
```diff
@@ -5,16 +5,16 @@ Commitment Discount Unit represents the provider-specified measurement unit indi
 The CommitmentDiscountUnit column adheres to the following requirements:
 
 * CommitmentDiscountUnit MUST be present in a FOCUS dataset when the provider supports [*commitment discounts*](#glossary:commitment-discount).
-* CommitmentDiscountUnit MUST be of type String, and the units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
-* The CommitmentDiscountUnit MUST be the same across all *rows* where *CommitmentDiscountQuantity* has the same [*CommitmentDiscountId*](#commitmentdiscountid).
-* CommitmentDiscountUnit MAY be null if [*CommitmentDiscountId*](#commitmentdiscountid) is not null and [*ChargeClass*](#chargeclass) is "Correction".
-* CommitmentDiscountUnit MUST NOT be null when *CommitmentDiscountId* is not null and *ChargeClass* is not "Correction".
+* CommitmentDiscountUnit MUST be of type String, and the units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [Unit Format](#unitformat) attribute.
```
Comment (ijurica) [inline specification/columns/commitmentdiscountunit.md:8]:
```suggestion
* CommitmentDiscountUnit MUST be of type String, and the units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
```

File: specification/columns/consumedunit.md (hunk 2 above)
Comment (ijurica) [inline specification/columns/consumedunit.md:8]:
```suggestion
* ConsumedUnit MUST be of type String, and the units of measure used in ConsumedUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
```

File: specification/columns/pricingunit.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
...
 
-The PricingUnit column MUST be present in a FOCUS dataset. This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
+The PricingUnit column MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset). This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [Unit Format](#unitformat) attribute.
```
Comment (ijurica) [inline specification/columns/pricingunit.md:5]:
```suggestion
The PricingUnit column MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset). This column MUST be of type String. It MUST NOT be null when [ChargeClass](#chargeclass) is not "Correction" and [ChargeCategory](#chargecategory) is "Usage" or "Purchase", MUST be null when ChargeCategory is "Tax", and MAY be null for all other combinations of ChargeClass and ChargeCategory. Units of measure used in PricingUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
```

File: specification/columns/consumedunit.md (hunk 2 above)
Comment (ijurica) [reply specification/columns/consumedunit.md]: suggestion addressed. resolving conversation.

**Comment by jpradocueva** [comment]
### Action Items from Maintainers' call on Oct 7:
- [ ] [TF1-#[589](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/589)] Maintainers: Validate the corrections and verify that references match the new format.

**Review by udam-f2 — APPROVED** [review]

**Review by jpradocueva — APPROVED** [review]
Ready to merge.

**Comment by jpradocueva** [comment]
Approved during the Members' call on Oct 8.

**Pull request merged**